a2aclient/jsonrpc_test.go

```diff
@@ -182,6 +182,62 @@ func TestJSONRPCTransport_GetTask(t *testing.T) {
 	}
 }
 
+func TestJSONRPCTransport_ListTasks(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		req := mustDecodeJSONRPC(t, r, "tasks/list")
+
+		resp := newResponse(
+			req,
+			json.RawMessage(
+				`{"tasks":[{
+				"kind":"task",
+				"id":"task-1",
+				"contextId":"ctx-1",
+				"status":{"state":"completed"}},
+				{"kind":"task",
+				"id":"task-2",
+				"contextId":"ctx-2",
+				"status":{"state":"working"}}], 
+				"total_size": 2, "page_size": 10, "next_page_token": "test-page-token"}`),
+		)
```

Comment:
nit: I'd reformat it to be more JSON-like:

```suggestion
		resp := newResponse(
			req,
			json.RawMessage(
				`{
					"tasks":[
						{"kind":"task", "id":"task-1", "contextId":"ctx-1", "status":{"state":"completed"}},
						{"kind":"task", "id":"task-2", "contextId":"ctx-2", "status":{"state":"working"}}
					],
					"total_size": 2,
					"page_size": 10,
					"next_page_token": "test-page-token"
				}`,
			),
		)
```